feat: implement dynamic filters for application types and status

29-29 opened the pull request:
### Pull Request Description

**Title:** feat: implement dynamic filters for application types and status

**Description:**
This pull request introduces dynamic filtering capabilities for application types and status within the application. The new feature allows users to easily filter and view applications based on their specific criteria, enhancing the overall user experience and improving data accessibility.

**Changes Made:**
- Implemented dynamic filters for application types.
- Added functionality to filter applications based on their current status.

<!-- GitButler Footer Boundary Top -->
---
This is **part 2 of 2 in a stack** made with GitButler:
- <kbd>&nbsp;2&nbsp;</kbd> #91 👈 
- <kbd>&nbsp;1&nbsp;</kbd> #90 
<!-- GitButler Footer Boundary Bottom -->

29-29 committed:
feat: implement dynamic filters for application types and status

fencim committed:
Merge branch 'develop' into applications-datatable-filter

fencim commented on src/pages/admin/ApplicationPage.vue:
```diff
@@ -3,11 +3,11 @@
         <Toolbar />
-        <q-tabs v-model="tab" dense active-color="primary">
+        <!-- <q-tabs v-model="tab" dense active-color="primary">
           <q-route-tab to="student" label="Student" />
           <q-route-tab to="adviser" label="Adviser" />
           <q-route-tab to="admin" label="Admin" />
-        </q-tabs>
+        </q-tabs> -->
```
you may want remove commented codes

fencim commented on src/components/admin/DataTable.vue:
```diff
@@ -75,22 +112,74 @@
+  color?: string;
+}
+
+const routePath = $route.path.split('/')[2];
```
reading this is vague, better to use `$route.params.type` which will require to change your route configuration. 
 ```suggestion
const routeApplicationType = $route.param.type as string;
```

fencim replied:
you need to update your route as follows without sub route to simplify it:
```ts
     {
        path: 'application/:type',
        name: 'application',
        component: () => import('pages/admin/ApplicationPage.vue'),
        meta: {
          requiresLogin: ['admin'],
          menu: 'Application',
          title: 'Application Page',
          icon: 'work_outline',
       },
```
removing the child routes

fencim reviewed (dismissed):
Your change is more simplified, but consider to implement suggestion in the comments.

fencim commented on src/pages/admin/ApplicationPage.vue:
```diff
@@ -3,11 +3,11 @@
           <q-route-tab to="admin" label="Admin" />
-        </q-tabs>
+        </q-tabs> -->
         <router-view />
```
You can remove this with the suggetion and replace it `<DataTable/>`
```suggestion
        <DataTable />
```

fencim commented on src/pages/admin/StudentAppPage.vue:
You may delete this file and the other similar Sub Vue Pages.

29-29 committed:
refactor: remove unused components from admin pages

29-29 committed:
feat: refactor application routes and update components

- Removes unnecessary child routes from the application and updates the main application route to accept a dynamic parameter. 
- Introduces a new DataTable component to replace the previous layout, enhancing the user interface. 
- Adds default parameter handling in the routing logic to streamline navigation.
